libs/ui/src/lib/fire-calculator/fire-calculator.service.ts

```diff
@@ -40,4 +40,36 @@ export class FireCalculatorService {
       totalAmount
     };
   }
+
+  public calculatePeriodsToRetire({
+    P,
+    PMT,
+    r,
+    totalAmount
+  }: {
+    P: number;
+    PMT: number;
+    r: number;
+    totalAmount: number;
+  }) {
+    if (r == 0) {
+      // No compound interest
+      return (totalAmount - P) / PMT;
+    } else if (totalAmount <= 0) {
+      return 0;
+    }
+
+    const periodInterest = new Big(r).div(this.COMPOUND_PERIOD);
+    const numerator1: number = Math.log10(
```

Comment:
Do you need to transform to `number` here?

Reply:
I didn't find a good library to take logarithms of big numbers. Probably it is not feasible to do for big numbers. That's why I needed to cast to a number here. For the numbers this is used for in our context, this should be fine though.

Reply:
Thanks for the clarification